base/ltfinal.dtx

```diff
@@ -1209,6 +1214,24 @@
 \cs_new_eq:NN \CaseSwitch \text_case_switch:nnnn
 \cs_new_eq:NN \DeclareCaseChangeEquivalent
   \text_declare_case_equivalent:Nn
+\NewDocumentCommand \DeclareLowercaseMapping { o m m }
+  {
+    \IFNoValueTF {#1}
+      { \text_declare_lowercase_mapping:nn {#2} {#3} }
+      { \text_declare_lowercase_mapping:nnn {#2} {#1} {#3} }
```

Comment:
... but I'm less sure about the order of arguments here: it might seems natural to put the locale in the middle in this case, but there may be other declarations that have a dependency on the locale and they many not naturally have "locale" in second place. My guess it is better to put it first in all such commands so that the logic is: "do XYZ"  or "for locale=abc do XYZ"  with XYZ having as many arguments as needed but in both cases in the same order.

Reply:
I only finalised the `expl3` layer yesterday so I think we are free to change - it does happen. I did wonder about the ordering, which would comes currently as
```latex
\text_declare_lowercase_mapping:nnn { "01F0 } { xx } { \v { j } }
```
but would if changed be
```latex
\text_declare_lowercase_mapping:nnn { xx } { "01F0 } { \v { j } }
```
I can live with either: the latter is closer to what `\text_lowercase:nn` does, but at the same time here with three arguments I could see a logic to having the codepoint first.

Reply:
as I said, my argument is that later on you may have other declarations that are locale specific and they may have more or less number of additional arguments so it seems more consistent to have the locale always first or for the global case "missing" rather than somewhere in the middle (or missing there). Or saying it differently are there a lot of things related to a code point and then subdivided by locale each or a lot of things related to a locale? I think the latter.

Reply:
another argument is that this matches order in the optional arg version

Reply:
OK, I'll adjust in `expl3`, do a release and adjust in the 2e PR